### ISSUE TITLE
Add an option to export forms and more…

### ISSUE THREAD

**Issue by planepix** [contao/contao#402]
**Description**  
As discussed at Contao Agenturtag it would be helpful to have an option to export defined forms and import them into another installation.

**Example**  
Add an export and import button in the form settings so setup can be easily exchanged between serveral installations.

**Comment by fritzmg** [comment]
Why just forms? This would be helpful for... well... _everything_ ;).

**Comment by contaoacademy** [comment]
@fritzmg Yes, it would be great for much more things! I would prefer permissions, site-structure and settings for responsive-images ;-)

**Comment by fritzmg** [comment]
I briefly talked with @aschempp at the developer meeting that I would like to have something similar to the `drush config-export/import` functionality of Drupal. This wouldn't just be an export - you should be able to synchronize a remote installation with that. This would require every database entry to have an UUID and ideally Contao would also internally only use those UUIDs.

**Comment by leofeyer** [comment]
What we have discussed at the Agenturtag was an option to export one or more records and to re-import them in another installation – similar to a clipboard. Since it is basically copy and paste, the imported records are always appended and do not require UUIDs.

**Comment by fritzmg** [comment]
I know that the pure export from one installation to another unrelated installation would be rather simple. But since we are discussing such a thing anyway, I would like to discuss the next evolution of this as well ;).

Sure - this would be something for the distant future. But if we are already implementing some sort of export functionality for database records, I'd like to keep the next step in mind (if we agree that we ever want this).

**Comment by MDevster** [comment]
We have a budget which we would like to put into a uuid concept for content elements and the page structure. We would like to sync multiple installations in either direction, or just certain elements or parts of the page structure. In an old news on contao.org I found a database model https://contao.org/en/news/contao-core-database-table-relations.html
Is there a current version of the model available?

**Comment by richardhj** [comment]
I have something called “Federated Sync” on my wishlist. I borrowed the term from Nextcloud, where you can aggregate the filesystem of different instances. 
For Contao, it should be possible to join, e.g., news archives from different Contao instances. I see the API being a requirement for this feature. And I guess the same is for the clipboard feature.

**Comment by operatorone** [comment]
I would really appreciate and support an extension which makes it possible to copy/paste/snyc between instances.

**Comment by frontendschlampe** [comment]
for syncing you can use syncCto ...

**Comment by operatorone** [comment]
Sure I can use syncCto, but it's more about coping  single or multiple elements from one system to another, for i.e. if you build up a new instance or extend an existing system by copying  modules, pages, articles etc. to the target instance it saves a lot of time. 

Of course, you always can argue to copy entries from one database to another, but this always brings up the ID problem.

To have an option (checkbox, link) to copy even single elements, which are already built, between two connected systems would be a great enhancement for all our workflows and would save a ton of time/money.

We all know the problems from working in a local instance and building up and style the backend and frontend elements. After that, you commit and push your work to the repo. And then? You build up the same elements in the target system again because the database differs. Ok, you can also build up the elements in the target system, make a dump and import it in your local environment. Yes, works, but during development you realize you forgot something, and you start over again. I think it's a bit exhausting.

I think, for the success of Contao, if this could be a core feature some time, this would have quite an impact on the developer community and even the regular customer, who does just content management will participate from this at a relaunch of a new website or if working with staging systems.

**Comment by Ernestopheles** [comment]
I am missing such a feature since long

**Comment by operatorone** [comment]
So let's build up a crowdfunding and get things going. 

So, we can start to save time.

**Comment by amenk** [comment]
Is this similar to https://github.com/contao-community-alliance/contao-clipboard-bundle ?
@operatorone How would you like to setup a crowfunding?

**Comment by operatorone** [comment]
@amenk 

Yes, it's kind of similar, but I don't know the full potential of the clipboard extension. 

I never started a crowdfunding campaign. Any ideas for a platform? Maybe https://www.startnext.com?

For the start, there has to be a feature list and an estimation of cost to get the budget right, I think.
